Docs/lambda-annotations-design.md

```diff
@@ -0,0 +1,358 @@
+{
+    if (!request.PathParameters.TryGetValue("x", out var xs))
+    {
+        return new APIGatewayProxyResponse
+        {
+            StatusCode = (int)HttpStatusCode.BadRequest
+        };
+    }
+    if (!request.PathParameters.TryGetValue("y", out var ys))
+    {
+        return new APIGatewayProxyResponse
+        {
+            StatusCode = (int)HttpStatusCode.BadRequest
+        };
+    }
+
+    var x = int.Parse(xs);
+    var y = int.Parse(ys);
+
+    return new APIGatewayProxyResponse
+    {
+        StatusCode = (int)HttpStatusCode.OK,
+        Body = (x + y).ToString(),
+        Headers = new Dictionary<string, string> { { "Content-Type", "text/plain" } }
+    };
+} 
+```
+
+Developers also have to keep a CloudFormation template that links to actual piece of code in sync. Every time a developer writes a new Lambda function they have to remember to update the CloudFormation template. If the user renames the method, class or namespace and they forget to make the corresponding changes to the CloudFormation template there is no error till the Lambda function is invoked in Lambda.
+
+
+### ASP.NET Core Lambda Functions
+
+The alternative for writing .NET Lambda functions for REST APIs is to use .NET�s ASP.NET Core web framework. This has been a popular feature of our .NET experience since the beginning. It allows developers to write REST APIs in a well known and documented framework. The style is idiomatic .NET using .NET attributes and reflection to map HTTP request data into corresponding to method parameters. Some of the features that are most appreciated with this approach are:
+
+* .NET Attributes for mapping route mapping and HTTP request data
+* Injecting middleware into request pipelines 
+    * A piece of code that runs before or after any request in the project
+* Integrated dependency injection
+    * A central location for registering all of the dependencies for the request in the project.
+* Authentication
+    * Define policies and add annotations to request paths that ensure the caller has the permissions for the policies.
+
+
+Here is an example of a ASP.NET Core REST controller doing similar logic to the low level example mention before.
+
+
+```
+[Route("[controller]")]
+public class MathController : ControllerBase
+{
+    [HttpGet("/plus/{x}/{y}")]
+    public int Plus(int x, int y)
+    {
+        return x + y;
+    }
+
+    [HttpGet("/subtract/{x}/{y}")]
+    public int Subtract(int x, int y)
+    {
+        return x - y;
+    }
+}
+```
+
+
+There are a few significant negative side effect for writing REST APIs for Lambda using ASP.NET Core. 
+
+* Significant impact to cold start due to size of the framework and reliance on reflection.
+* The common features of ASP.NET Core for REST APIs are supported in Lambda but a significant number of the features are not and there is no clear indication what is and isn�t supported.
```

Comment:
**significant number of the features** can we be clear what features?

Reply:
Added a few examples.